### DISCUSSION TITLE
rtsps source ubiquitti unifi protect

### DISCUSSION THREAD

**Discussion by mabunixda** [bluenviron/mediamtx#652] [Questions] [answered]
Hi,

i try to convert a rtsp source - in the mean time as rtsps only available - from unifi protect into a mpjeg stream to get availability of this stream in another software automation. 

I configured source to rtsps://<unifi-protect-url> and sourceFingerprint , based on the openssl commands - in the configuration settings and want to use the container deployment. 

But I get following output:
```
2021/10/19 09:04:36 I [0/0] [path cam1] [rtsp source] ready
2021/10/19 09:04:36 I [0/0] [HLS] [conn 172.17.0.1:63108] GET /cam1/index.m3u8
2021/10/19 09:04:36 I [0/0] [HLS] [muxer cam1] created
2021/10/19 09:04:36 I [0/1] [HLS] [muxer cam1] is converting into HLS
2021/10/19 09:04:36 I [0/1] [HLS] [conn 172.17.0.1:63108] GET /cam1/stream.m3u8
2021/10/19 09:04:36 W [0/1] [HLS] [muxer cam1] unable to decode audio track: invalid AU-headers-length (15992)
2021/10/19 09:04:36 W [0/1] [HLS] [muxer cam1] unable to decode audio track: invalid AU-headers-length (7839)
2021/10/19 09:04:36 W [0/1] [HLS] [muxer cam1] unable to decode audio track: invalid AU-headers-length (55164)
2021/10/19 09:04:36 W [0/1] [HLS] [muxer cam1] unable to decode audio track: invalid AU-headers-length (4509)
2021/10/19 09:04:36 W [0/1] [HLS] [muxer cam1] unable to decode audio track: AU-index field is not zero
2021/10/19 09:04:36 W [0/1] [HLS] [muxer cam1] unable to decode audio track: AU-index field is not zero
```

Is the stream broken itself or did I miss some config options?

**Comment by mabunixda** [comment, marked as answer]
It was related to [discussion 454](https://github.com/aler9/rtsp-simple-server/discussions/454). Increasing the ```readBufferSize``` solved also the problem!

**Reply by jplitza** [reply]
readBufferSize no longer exists since 98c6cd4, so what do I do now?

**Reply by aler9** [reply]
@jplitza the reason why readBufferSize was removed is that packets with a size greater than 1500 can't be routed correctly with all supported transport protocols (TCP, UDP, Multicast).

Gortsplib can receive packets of any size but can't send packets with a size greater with 1500.

MediaMTX detects oversized packets and remuxes them into smaller ones, but this currently works only if the codec is H265 or H264. Anyway, the strategy is to remux packets instead of allowing invalid ones to flow through the server.

If you are experiencing difficulties with a device that sends oversized packets, open an issue in MediaMTX and provide details that allow to replicate the issue.

[a comment by an automated account is not shown]
